docs/source/development/documentation.rst

```diff
@@ -1,3 +1,127 @@
-.. _documentation:
+Adding to the Documentation
+============================
 
-.. mdinclude:: documentation.md
\ No newline at end of file
+Documentation is an integral part of every collaborative software project. Good documentation not only encourages users of the package to try out different functionalities, but it also makes maintaining and expanding code significantly easier. If you want to improve ProbNum's documentation or learn how to write documentation for your newly implemented functionality, keep reading.
+
+Docstrings
+***********
+
+The main form of documentation are `docstrings`, multi-line comments beneath a class or function definition with a specific syntax, which detail its functionality. This package uses the `NumPy docstring format <https://numpydoc.readthedocs.io/en/latest/format.html#numpydoc-docstring-guide>`_. As a rule, all functions which are exposed to the user *must* have appropriate docstrings. Below is an example of a docstring for a probabilistic numerical method.
+
+.. code-block:: python
+
+	def problinsolve(A, b, A0=None, Ainv0=None, x0=None, assume_A="sympos", maxiter=None, atol=10 ** -6, rtol=10 ** -6,
+	                 callback=None, **kwargs):
+	    """
+	    Infer a solution to the linear system :math:`A x = b` in a Bayesian framework.
+
+	    Probabilistic linear solvers infer solutions to problems of the form
+
+	    ...
+
+	    Parameters
+	    ----------
+	    A : array_like or LinearOperator, shape=(n,n)
+	        A square matrix or linear operator.
+
+	    ...
+
+	    Returns
+	    -------
+	    x : RandomVariable, shape=(n,) or (n, nrhs)
+	        Approximate solution :math:`x` to the linear system. Shape of the return matches the shape of ``b``.
+
+			...
+
+	    Raises
+	    ------
+	    ValueError
+	        If size mismatches detected or input matrices are not square.
+
+	    Notes
+	    -----
+	    For a specific class of priors the probabilistic linear solver recovers the iterates of the conjugate gradient
+
+	    ...
+
+	    References
+	    ----------
+	    .. [1] Wenger, J. and Hennig, P., Probabilistic Linear Solvers for Machine Learning, 2020
+
+	    ...
+
+	    See Also
+	    --------
+	    bayescg : Solve linear systems with prior information on the solution.
+
+	    Examples
+	    --------
+	    >>> import numpy as np
+	    >>> np.random.seed(1)
+	    >>> n = 20
+	    >>> A = np.random.rand(n, n)
+	    >>> A = 0.5 * (A + A.T) + 5 * np.eye(n)
+	    >>> b = np.random.rand(n)
+	    >>> x, A, Ainv, info = problinsolve(A=A, b=b)
+	    >>> print(info["iter"])
+	    10
+	    """
+
+
+**General Rules**
+
+- Cover :code:`Parameters`, :code:`Returns`, :code:`Raises` and :code:`Examples` at every publicly visible docstring---in that order.
+- Examples are tested via doctest. Ensure :code:`doctest` does not fail.
```

Comment:
Do we actually run doctests right now? Is that included in pytest by default?

Reply:
In the `setup.cfg` file there is an option configuring `pytest`. I also added `--doc-test` to the `tox.ini` now. This generates this output indicating that doctests are run:
```bash
src/probnum/diffeq/ode/ivp.py::src.probnum.diffeq.ode.ivp.IVP PASSED     [  0%]
src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py::src.probnum.diffeq.odefiltsmooth.odefiltsmooth.probsolve_ivp PASSED [  0%]
src/probnum/linalg/linearsolvers/linearsolvers.py::src.probnum.linalg.linearsolvers.linearsolvers.problinsolve PASSED [  1%]
src/probnum/linalg/linops/linearoperators.py::src.probnum.linalg.linops.linearoperators.LinearOperator PASSED [  1%]
src/probnum/linalg/linops/linearoperators.py::src.probnum.linalg.linops.linearoperators.aslinop PASSED [  2%]
src/probnum/prob/randomvariable.py::src.probnum.prob.randomvariable.asrandvar PASSED [  2%]
src/probnum/prob/distributions/dirac.py::src.probnum.prob.distributions.dirac.Dirac PASSED [  2%]
src/probnum/prob/distributions/normal.py::src.probnum.prob.distributions.normal.Normal PASSED [  3%]
src/probnum/quad/polynomial/clenshawcurtis.py::src.probnum.quad.polynomial.clenshawcurtis.ClenshawCurtis PASSED [  3%]

```

Reply:
Cool, thanks!